content/en/blog/2020/dns-proxy/index.md

```diff
@@ -0,0 +1,252 @@
+instance IPs that can then be directly accessed by the
+application. For example, consider the two service entries below,
+pointing to two different AWS RDS services:
+
+{{< text yaml >}}
+apiVersion: networking.istio.io/v1alpha3
+kind: ServiceEntry
+metadata:
+  name: db1
+  namespace: ns1
+spec:
+  hosts:
+  - mysql–instance1.us-east-1.rds.amazonaws.com
+  ports:
+  - name: mysql
+    number: 3306
+    protocol: TCP
+  resolution: DNS
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: ServiceEntry
+metadata:
+  name: db2
+  namespace: ns1
+spec:
+  hosts:
+  - mysql–instance2.us-east-1.rds.amazonaws.com
+  ports:
+  - name: mysql
+    number: 3306
+    protocol: TCP
+  resolution: DNS
+{{< /text >}}
+
+The sidecar has a single listener on `0.0.0.0:3306` that looks up the
+IP address of `mysql-instance1.us-east1.rds.amazonaws.com` from public
+DNS servers and forwards traffic to it. It cannot route traffic to
+`db2` as it has no way of distinguishing whether traffic arriving at
+`0.0.0.0:3306` is bound for `db1` or `db2`. The only way to accomplish
+this is to set the resolution to `NONE` causing the sidecar to
+_blindly forward any traffic_ on port `3306` to the original IP
+requested by the application. This is akin to punching a hole in the
+firewall allowing all traffic to port `3306` irrespective of the
+destination IP. To get traffic flowing, you are now forced to
+compromise on the security posture of your system.
+
+### Resolving DNS for services in remote clusters
+
+The DNS limitations of a multicluster mesh are well known. Services in
+one cluster cannot lookup the IP addresses of services in other
+clusters, without clunky workarounds such as creating stub services in
+the caller namespace.
+
+## Taking control of DNS
+
+All in all, DNS has been a thorny issue in Istio for a while. It was
+time to slay the beast. We (the Istio networking team) decided to
+tackle the problem once and for all in a way that is completely
+transparent to you, the end user. Our first attempt involved utilizing
+Envoy's DNS proxy. It turned out to be very unreliable, and
+disappointing overall due to the general lack of sophistication in
+c-ares library, the DNS library used by Envoy. Determined to solve the
+problem, we decided to implement the DNS proxy in the Istio sidecar
+agent, written in Go. We were able to optimize the implementation to
+handle all the scenarios that we wanted to tackle without compromising
+on scale and stability. The Golang DNS packages we use are the same as
+those used by scalable DNS implementations such as CoreDNS, Consul,
+Mesos, among others.
+
+Starting with Istio 1.8, the Istio agent on the sidecar will ship with
+a caching DNS proxy, programmed dynamically by Istiod. Istiod pushes
+the hostname to IP address mappings for all the services that the
+application may access based on the Kubernetes services and Service
+Entries in the cluster. DNS lookup queries from the application are
```

Comment:
```suggestion
application may access based on the Kubernetes services and service
entries in the cluster. DNS lookup queries from the application are
```
How does it determine the accessible services in the VM case, where there is no cluster?

Reply:
The VM service is just like any other service in the mesh. It can access services in same or other namespaces subject to the Sidecar crd.